libs/langchain_v1/langchain/agents/middleware/human_in_the_loop.py

```diff
@@ -1,135 +1,255 @@
-    HumanInterruptConfig,
-    HumanResponse,
-)
+from typing import Any, Literal
+
+from langchain_core.messages import AIMessage, ToolCall, ToolMessage
 from langgraph.types import interrupt
+from typing_extensions import NotRequired, TypedDict
 
-from langchain.agents.middleware._utils import _generate_correction_tool_messages
 from langchain.agents.middleware.types import AgentMiddleware, AgentState
 
-ToolInterruptConfig = dict[str, HumanInterruptConfig]
+
+class HumanInTheLoopConfig(TypedDict):
+    """Configuration that defines what actions are allowed for a human interrupt.
+
+    This controls the available interaction options when the graph is paused for human input.
+
+    Attributes:
+        allow_accept: Whether the human can approve the current action without changes
+        allow_respond: Whether the human can reject the current action with feedback
+        allow_edit: Whether the human can approve the current action with edited content
+    """
+
+    allow_accept: NotRequired[bool]
+    allow_edit: NotRequired[bool]
+    allow_respond: NotRequired[bool]
+
+
+class ActionRequest(TypedDict):
+    """Represents a request with a name and arguments.
+
+    Attributes:
+        action: The type or name of action being requested (e.g., "add_numbers")
+        args: Key-value pairs of arguments needed for the action (e.g., {"a": 1, "b": 2})
+    """
+
+    action: str
+    args: dict
+
+
+class HumanInTheLoopRequest(TypedDict):
+    """Represents an interrupt triggered by the graph that requires human intervention.
+
+    Attributes:
+        action_request: The specific action being requested from the human
+        config: Configuration defining what response types are allowed
+        description: Optional detailed description of what input is needed
+    Example:
+        ```python
+        # Extract a tool call from the state and create an interrupt request
+        request = HumanInterrupt(
+            action_request=ActionRequest(
+                action="run_command",  # The action being requested
+                args={"command": "ls", "args": ["-l"]},  # Arguments for the action
+            ),
+            config=HumanInTheLoopConfig(
+                allow_accept=True,  # Allow approval
+                allow_respond=True,  # Allow rejection with feedback
+                allow_edit=False,  # Don't allow approval with edits
+            ),
+            description="Please review the command before execution",
+        )
+        # Send the interrupt request and get the response
+        response = interrupt([request])[0]
+        ```
+    """
+
+    action_request: ActionRequest
+    config: HumanInTheLoopConfig
+    description: str | None
+
+
+class AcceptPayload(TypedDict):
+    """Response when a human approves the action."""
+
+    type: Literal["accept"]
+
+
+class ResponsePayload(TypedDict):
+    """Response when a human rejects the action."""
+
+    type: Literal["response"]
+
+    args: NotRequired[str]
```

Comment:
Could you add a doc-string to explain that if provided the `args` is a string that contains the message from the user explaining why the action is being rejected